### PR TITLE
Misc memory leaks fixes

### PR DESCRIPTION
**Pull request by jeromelaban**
## PR Type

What kind of change does this PR introduce?
- Bugfix

## What is the new behavior?

- Ensure that `ItemCollection` is unregistering from `ItemSource`
- Use weak references for `ElementNameSubject` and `TargetPropertyPath`
- Pin the generated instance for `DataTemplate` holder for `x:Name` to allow for `ElementNameSubject` and `TargetPropertyPath` weak references.
- Improve the memory leak runtime tests by tracking children instances and properties
- Fix FrameworkTemplate resource owner leak (explicit weak propagation through FrameworkTemplate)
- Move iOS RuntimeTests to their own job to reduce timeout impact

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Review by mauadi — APPROVED** [review]
imagentleman

File: src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/Button/Buttons.xaml.cs [hunk 1]
```diff
@@ -19,7 +19,8 @@ public Buttons()
 		{
 			this.InitializeComponent();
 
-			DataContext = _viewModel = new ButtonsViewModel();
+			Loading += (s, e) => DataContext = _viewModel = new ButtonsViewModel();
```
Comment (davidjohnoliver) [inline src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/Button/Buttons.xaml.cs:22]: Is this a workaround?

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/ResourceOwner.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+﻿namespace Uno.UI.SourceGenerators.XamlGenerator
+{
+	/// <summary>
+	/// Resource owner variable definition
```
Comment (davidjohnoliver) [inline src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/ResourceOwner.cs:4]: Would be nice if you can expand this - it's not very obvious why we need Yet Another notion of scope.

**Review by davidjohnoliver — APPROVED** [review]

**Review by kazo0 — APPROVED** [review]

**Review by ajpinedam — APPROVED** [review]

File: src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/Button/Buttons.xaml.cs (hunk 1 above)
Comment (jeromelaban) [reply src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/Button/Buttons.xaml.cs:22]: Somehow. If the DataContext stays set, then the event registrations are kept alive, causing the view to stay alive.

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/ResourceOwner.cs (hunk 2 above)
Comment (jeromelaban) [reply src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/ResourceOwner.cs:4]: Will do in a subsequent PR

**Pull request merged**